### ISSUE TITLE
v 1.3.2 Object is not set to an instance of an object, but not in v 1.3.0

### ISSUE THREAD

**Issue by adcorduneanu** [BenMorris/NetArchTest#103]
On the project, I'm working on I tried upgrading the existing version 1.3.0 to 1.3.2, but had to revert, because there is a bug in 1.3.2, which doesn't let me get an instance of Conditions via Should, from my PredicateList.

The old Object is not set to an instance of an object kicked in...

What it should do? 
Get all usages of a specified dependency, in the given assemblies.

![image](https://user-images.githubusercontent.com/5188847/169501980-92406c1e-c739-489d-af2b-173f0f6a31ff.png)

**Comment by BenMorris** [comment]
I've not been able to replicate this and nobody else has reported this issue. 

From what I can tell there's been no change to the [Should()](https://github.com/BenMorris/NetArchTest/blob/367d3baea683a3a1110b744d301ae40b20985dcb/src/NetArchTest.Rules/PredicateList.cs#L33) method for some time - it still returns an instance of the [Conditions](https://github.com/BenMorris/NetArchTest/blob/master/src/NetArchTest.Rules/Conditions.cs) class.

**Comment by adcorduneanu** [comment]
After 2 years I've decided to get to the bottom of this as I remained with 1.3.0 for the last 2 years. 

In one of my classes, I had a constant that pointed to a string that intentionally had the null value - to test some extension methods.

The code that verifies the constants somehow can't properly resolve it, and the field.Constant.ToString(), fails as the `field.Constant` is null.

![Image](https://github.com/user-attachments/assets/a5c8e231-cd28-4ea1-be92-ac946ad90221)

https://github.com/BenMorris/NetArchTest/blob/a761f63b694a6facd4be01fbc8fe2ae26579572e/src/NetArchTest.Rules/Dependencies/TypeDefinitionCheckingContext.cs#L110

It's a pretty "niche" use case, so it's tough to find.
